src/binary/decimal.rs

```diff
@@ -137,13 +138,18 @@ mod binary_decimal_tests {
     use super::*;
     use rstest::*;
 
-    /// This test ensures that we implement PartialEq correctly for the special
+    /// This test ensures that we implement [PartialEq] and [IonEq] correctly for the special
     /// decimal value 0d0.
     #[test]
     fn decimal_0d0_is_a_special_zero_value() {
         assert_eq!(DECIMAL_POSITIVE_ZERO, Decimal::new(0, 0));
-        assert_ne!(DECIMAL_POSITIVE_ZERO, Decimal::new(0, 10));
-        assert_ne!(DECIMAL_POSITIVE_ZERO, Decimal::new(0, 100));
+        assert!(DECIMAL_POSITIVE_ZERO.ion_eq(&Decimal::new(0, 0)));
+
+        assert_eq!(DECIMAL_POSITIVE_ZERO, Decimal::new(0, 10));
+        assert!(!DECIMAL_POSITIVE_ZERO.ion_eq(&Decimal::new(0, 10)));
+
+        assert_eq!(DECIMAL_POSITIVE_ZERO, Decimal::new(0, 100));
+        assert!(!DECIMAL_POSITIVE_ZERO.ion_eq(&Decimal::new(0, 100)));
```

Comment:
🗺️ Unit tests that previously relied on `==`/`eq()` now use `ion_eq()`.

`ion_eq()` is stricter than `eq`, so if we're making an assertion about two values being `IonEq` then we can also assert that they are `PartialEq`. Likewise, if we say they're _not_ `IonEq`, we can also assert that they are not `PartialEq`.